pkgs/applications/version-management/mercurial/default.nix

```diff
@@ -22,6 +22,8 @@ in python3Packages.buildPythonApplication rec {
   buildInputs = [ makeWrapper docutils unzip ]
     ++ lib.optionals stdenv.isDarwin [ ApplicationServices ];
 
+  propagatedBuildInputs = [ hg-git dulwich ];
```

Comment:
You shouldn't add this here; hg-git should use the mercurial.withExtensions mechanism.